frontend/src/components/Pages/InspectionPage/ScheduleMissionDialog.tsx

```diff
@@ -58,6 +83,21 @@ export const ScheduleMissionDialog = (props: IProps): JSX.Element => {
         return () => clearInterval(id)
     }, [props.refreshInterval])
 
+    useEffect(() => {
+        const id = setInterval(() => {
+            let unscheduledMissions: CondensedMissionDefinition[] = []
+
+            props.missions.forEach((mission) => {
+                if (Object.keys(props.scheduledMissions).includes(mission.id) && props.scheduledMissions[mission.id])
+                    setIsAlreadyScheduled(true)
+                else unscheduledMissions = unscheduledMissions.concat([mission])
+            })
+
+            setUnscheduledMissions(unscheduledMissions)
+        }, props.refreshInterval)
+        return () => clearInterval(id)
+    }, [props.refreshInterval])
```

Comment:
What do you think about making this a function that checks if there are already scheduled missions when the schedule button is pressed?

Reply:
I like it, will change